### PR TITLE
feat(highlight): reduced exposed interface of Highlighter module

### PR DESCRIPTION
**Pull request by mukiienko**
BREAKING CHANGE: changed HighlighterCore API

### CODE REVIEW

**Comment by antipin** [comment]
This commit message will not be squashed, so it should be more meaningful. Something like
```
feat(highlight): reduced exposed interface of Highlighter moduile
```

File: src/packages/Highlighter/Node/HighlighterNode.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import randomcolor from 'randomcolor';
-import { HighlighterCore } from '../core';
+import { HighlighterCore } from '../Core';
```
Comment (antipin) [inline src/packages/Highlighter/Node/HighlighterNode.ts:2]: Filename and classname do not match folder name
`/Node/HighlighterNode.ts` --> `/HighlighterNode/HighlighterNode.ts` 

Same for 
`Core/HighlighterCore.ts` --> `HighlighterCore/HighlighterCore.ts`

Comment (mukiienko) [reply src/packages/Highlighter/Node/HighlighterNode.ts:2]: Done!

File: src/index.ts [hunk 2]
```diff
@@ -2,15 +2,12 @@
...
     HighlighterCoreOptions,
     HighlighterCoreAccuracy,
-} from './packages/@highlighter/core';
-export { HighlighterControl, HighlightControlScrollArgs } from './packages/@highlighter/control';
-export {
-    HighlighterNode,
-    HighlighterNodeOptions,
-    HighlighterNodeRenderer,
-} from './packages/@highlighter/node';
+} from './packages/Highlighter/Core';
```
[the remark this answers is not in the document]
Comment (mukiienko) [reply src/index.ts:8]: Good point!

File: src/components/Highlighter/GroupHighlighter/GroupHighlighter.tsx [hunk 3]
```diff
@@ -12,113 +11,33 @@ interface Props {
...
-    const {
-        children,
-        searchTerms,
-        accuracy,
-        ignoreDiacritics,
-        ignoreCase,
-        onComplete,
-        highlightRenderer,
-    } = props;
+    const { children, searchTerms, highlighterCoreOptions, onComplete } = props;
```
[the remark this answers is not in the document]
Comment (mukiienko) [reply src/components/Highlighter/GroupHighlighter/GroupHighlighter.tsx:25]: `StringHighlighter`-  now is used in Reach for highlighting result.
`ReactElementHighlighter` - I think that component doesn't have practical usage. The main idea was to show how `/packages/Highlighter` can be integrated and how it works. We can say `ReactElementHighlighter` is an example component. So I think to combine these components to one component is not a good idea, at least for now. What do you think?

[the remark this answers is not in the document]
Comment (antipin) [reply src/components/Highlighter/GroupHighlighter/GroupHighlighter.tsx]: Ok, I see now. Ok, let's keep them separate

**Pull request merged**